opentelemetry/src/exporter/trace/mod.rs

```diff
@@ -10,28 +11,25 @@ use http::Request;
 use serde::{Deserialize, Serialize};
 #[cfg(all(feature = "http", feature = "reqwest"))]
 use std::convert::TryInto;
-use std::error::Error;
-use std::fmt::{Debug, Display};
+use std::fmt::Debug;
+#[cfg(feature = "http")]
+use std::fmt::{Display, Formatter};
 use std::sync::Arc;
 use std::time::SystemTime;
 
 pub mod stdout;
 
 /// Describes the result of an export.
-pub type ExportResult = Result<(), Box<dyn std::error::Error + Send + Sync + 'static>>;
-
-/// Timed out when exporting spans to remote
-#[derive(Debug, Default)]
-pub struct ExportTimedOutError {}
+pub type ExportResult = Result<(), TraceError>;
 
-impl Display for ExportTimedOutError {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str("export timed out")
+/// Marker trait for errors returned by exporters
+pub trait ExportError: std::error::Error + Send + Sync + 'static {
```

Comment:
I wonder if this should be moved to the parent module, i.e. should metrics / trace / log export errors be basically the same, or do they all need different export errors?